openmc/mesh.py

```diff
@@ -1347,6 +1347,70 @@ def __repr__(self):
             string += fmt.format('\tZ Max:', '=\t', self._z_grid[-1])
         return string
 
+    def get_indices_at_coords(self, x: float, y: float, z: float) -> Tuple[int]:
+        """Finds the index of the mesh voxel at the specified x,y,z coordinates.
+
+        Parameters
+        ----------
+        x : float
+            The x axis value
+        y : float
+            The y axis value
+        z : float
+            The z axis value
+
+        Returns
+        -------
+        Tuple[int]
+            The r, phi, z indices
+
+        """
+        r_value_from_origin = sqrt((x-self.origin[0])**2 + (y-self.origin[1])**2)
+
+        r_grid_values = np.array(self.r_grid)
+
+        if r_value_from_origin < r_grid_values[0] or r_value_from_origin > r_grid_values[-1]:
+            raise ValueError(
+                f'The specified x, y ({x}, {y}) combine to give an r value of '
+                f'{r_value_from_origin} from the origin of {self.origin}.which is outside the origin absolute r grid values '
+                f'{r_grid_values}.'
+            )
+
+        r_index = np.argmax(r_grid_values > r_value_from_origin) - 1
```

Comment:
`np.searchsorted` would be more appropriate than `np.argmax`, and should in theory be faster